### PR TITLE
Changes to Scheduling docs

### PR DESCRIPTION
**Pull request by poornimanayar**
Hi,

I have added some more information to the scheduling docs. I have added some notes regarding the `RecurringTaskBase` and some information about events available in `BackgroundTaskRunner` class.

Poornima

### CODE REVIEW

File: Reference/Scheduling/index.md [hunk 1]
```diff
@@ -96,6 +96,49 @@ namespace Umbraco.Web.UI
 }
 
 ```
+### RecurringTaskBase
```
Comment (sofietoft) [inline Reference/Scheduling/index.md:99]:
```suggestion
## RecurringTaskBase
```

File: Reference/Scheduling/index.md [hunk 2]
```diff
@@ -96,6 +96,49 @@ namespace Umbraco.Web.UI
...
+
+This class provides the base class for any recurring task. You can override the *PerformRun* method to implement the class. Tasks can also be run asynchronously. In this case the property *IsAsync* must be overridden and set to false and the *PerformRunAsync* must be overridden to implement the class.
+
+### BackgroundTaskRunner Events
```
Comment (sofietoft) [inline Reference/Scheduling/index.md:103]:
```suggestion
## BackgroundTaskRunner Events
```

File: Reference/Scheduling/index.md [hunk 3]
```diff
@@ -96,6 +96,49 @@ namespace Umbraco.Web.UI
 }
 
 ```
+### RecurringTaskBase
+
+This class provides the base class for any recurring task. You can override the *PerformRun* method to implement the class. Tasks can also be run asynchronously. In this case the property *IsAsync* must be overridden and set to false and the *PerformRunAsync* must be overridden to implement the class.
```
Comment (sofietoft) [inline Reference/Scheduling/index.md:101]:
```suggestion
This class provides the base class for any recurring task. You can override the `PerformRun` method to implement the class. Tasks can also be run asynchronously. In this case, the property `IsAsync` must be overridden and set to false and the `PerformRunAsync` must be overridden to implement the class.
```

File: Reference/Scheduling/index.md [hunk 4]
```diff
@@ -96,6 +96,49 @@ namespace Umbraco.Web.UI
...
+
+<table>
+    <tr>
+        <th>Event</th>
+        <th>Signature</th>
+        <th>Description</th>
+    </tr>
+    <tr>
+        <td>TaskStarting</td>
+        <td>(BackgroundTaskRunner<IBackgroundTask> sender, TaskEventArgs<IBackgroundTask> e)</td>
```
Comment (sofietoft) [inline Reference/Scheduling/index.md:115]:
```suggestion
        <td>`(BackgroundTaskRunner<IBackgroundTask> sender, TaskEventArgs<IBackgroundTask> e)`</td>
```

File: Reference/Scheduling/index.md [hunk 5]
```diff
@@ -96,6 +96,49 @@ namespace Umbraco.Web.UI
...
+        <td>
+        Raised when a task starts
+        </td>
+    </tr>
+    <tr>
+        <td>TaskCompleted</td>
+        <td>(BackgroundTaskRunner<IBackgroundTask> sender, TaskEventArgs<IBackgroundTask> e)</td>
```
Comment (sofietoft) [inline Reference/Scheduling/index.md:122]:
```suggestion
        <td>`(BackgroundTaskRunner<IBackgroundTask> sender, TaskEventArgs<IBackgroundTask> e)`</td>
```

File: Reference/Scheduling/index.md [hunk 6]
```diff
@@ -96,6 +96,49 @@ namespace Umbraco.Web.UI
...
+        <td>
+        Raised when a task has completed
+        </td>
+    </tr>
+    <tr>
+        <td>TaskError</td>
+        <td>(BackgroundTaskRunner<IBackgroundTask> sender, TaskEventArgs<IBackgroundTask> e)</td>
```
Comment (sofietoft) [inline Reference/Scheduling/index.md:129]:
```suggestion
        <td>`(BackgroundTaskRunner<IBackgroundTask> sender, TaskEventArgs<IBackgroundTask> e)`</td>
```

File: Reference/Scheduling/index.md [hunk 7]
```diff
@@ -96,6 +96,49 @@ namespace Umbraco.Web.UI
...
+        <td>
+        Raised when a task throws error
+        </td>
+    </tr>
+    <tr>
+        <td>TaskCancelled</td>
+        <td>(BackgroundTaskRunner<IBackgroundTask> sender, TaskEventArgs<IBackgroundTask> e)</td>
```
Comment (sofietoft) [inline Reference/Scheduling/index.md:136]:
```suggestion
        <td>`(BackgroundTaskRunner<IBackgroundTask> sender, TaskEventArgs<IBackgroundTask> e)`</td>
```

**Comment by sofietoft** [comment]
Apologies for the delay on this one @poornimanayar !

We've discussed it at our bi-weekly Docs team meeting, and we have a few suggestions.
The table you've put in - the one explaining the `BackgroundTaskRunner events` - we feel it would be better to link to the API docs: https://our.umbraco.com/apidocs/v8/csharp/api/Umbraco.Web.Scheduling.BackgroundTaskRunner-1.html#events ...  

And perhaps -  it would be awesome with some examples of usage of these events!

**Comment by sofietoft** [comment]
Hi there!
I have an important message for you, regarding this PR:

Yesterday we tried to do some cleaning up and force-pushed to the master branch which now causes some problems with existing forks and local clones. 
We have fixed up all the open PRs so that they are still mergeable now.

With that said, if you've been working on a **fork of the repo** or in a **local clone**, those will not work anymore.
If you wish to push any new changes to this PR, you will need to delete your local clone or fork and create fresh ones.

If you have any local work in progress, then you copy it over from your old fork or clone, following these steps:

**Forks**
1. In the old clone of the old fork, switch to the branch you were working on
2. In the new clone of the new fork, create a new branch with the same name as the old one
3. Copy all the affected files from the old clone to the new clone
4. Repeat steps 1-3 if you've been working with multiple branches
5. You can now safely delete your old fork

**Clones**
1. In the old clone, switch to the branch you were working on
2. Create and checkout a new branch on the new clone, using the same name as the old clone
3. Copy all the affected files from the old clone to the new clone
4. Repeat steps 1-3 if you've been working with multiple branches
5. Delete the old clone of the docs repo

And with that, you should be all set once more.

I do apologize for the inconvenience, and please let me know if you have any questions!
/Sofie

**Comment by poornimanayar** [comment]
Hi @sofietoft ,

Should I totally delete my fork and create a new fork after backing up my changes? Is it still `master` branch we are using as starting point for branches or `main`?

I am a bit confused with the instructions around Forks as I cant re-fork without deleting my old fork. So I have to back up my changes , delete my fork, re-fork and then add again

Poornima

**Comment by sofietoft** [comment]
Hi! Good question.

So, you'll need to backup and then remove your old branch of the UmbacoDocs repo.
Then, you can create a new fork, and add your changes to that one.

Might need to create a new PR on this, but let's see.
The `master` branch has been renamed to `main` - your new fork should default to that.

I hope that answers your questions :)

**Comment by poornimanayar** [comment]
Also, do you have any ideas for examples? I can write to logs as an example which is something we often do. 

Poornima Nayar

**Comment by sofietoft** [comment]
Ah, sure! Let me just run it by my team - I'm sure they might have some ideas.
Writing to logs seems like a really good option though!

**Comment by poornimanayar** [comment]
PR with amends on https://github.com/umbraco/UmbracoDocs/pull/2660

**Pull request closed without merging**